added section on releasing event handlers on env object (#1452)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/concepts/developer-guide.md

<!-- is the Ref link enough, or link to a long section in regular docs? -->

## Event handlers on the environment object

If any of your app's event handlers on the [environment object][CreateCoreWebView2Environment] hold a reference to the environment object, and the app simply releases the reference to the environment and event handlers without removing the event handlers, there might be a circular reference between the environment object and handler objects, which will leak memory.

To prevent such a memory leak:
*  For any added event handler, remove the event handler before releasing the environment object.

> *  Avoid holding a reference to the environment object in the event handlers. The handler can access the environment object from the sender argument of the "event completed" callback.

Reviewer: ```suggestion
*  Avoid holding a reference to the environment object in the event handlers. The handler can access the environment object from the `sender` argument of the "event completed" callback.
```

> *  Avoid holding a reference to the environment object in the event handlers. The handler can access the environment object from the `sender` argument of the "event completed" callback.

Reviewer: ```suggestion
*  Avoid holding a reference to the environment object in an event handler.  Instead, the event handler can access the environment object from the `sender` argument of the "event completed" callback.
```

[WebView2ProcessFailedEvent]: /microsoft-edge/webview2/reference/win32/icorewebview2processfailedeventargs "ICoreWebView2ProcessFailedEventArgs | Microsoft Docs"

[MicrosoftedgeinsiderDownload]: https://www.microsoftedgeinsider.com/download "Download Microsoft Edge Insider Channels"

> [CreateCoreWebView2Environment]: /microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.961.33#createcorewebview2environment "CreateCoreWebView2Environment | Microsoft Docs"

Reviewer: ```suggestion
[CreateCoreWebView2Environment]: /microsoft-edge/webview2/reference/win32/webview2-idl#createcorewebview2environment "CreateCoreWebView2Environment | Microsoft Docs"
```
Doc build says:
Line 76, Column 44: [Suggestion-preserve-view-not-set] Did you mean to link to a specific version? It's usually better not to specify a version, so the link always goes to the canonical version. If you really need to link to a specific version, add &preserve-view=true to the URL: /microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.961.33#createcorewebview2environment. Otherwise, remove the view parameter.
